web3c/oasis.js

```diff
@@ -312,6 +474,37 @@ class Oasis {
     };
   }
 
+  _setupInvoke(options = {}) {
+    // for invoke to work, we need to keep track of which subscriptions are currently
+    // available to avoid recreating subscriptions.
+    options.oasis = this;
+    this._invokeSubscription = new InvokeSubscription(options);
+
+    this._executeSend = async (subscription, send, contract, sendArgs) => {
+      const promise = this._invokeSubscription.createResolvablePromise();
+      const address = contract.address ? contract.address : contract.options.address;
+      await send.apply(contract, sendArgs)
+        .on('transactionHash', hash => {
+          this._invokeSubscription.pushExpectedTransaction(subscription.fromAddress, {
```

Comment:
Does this mean we are successfully subscribing *after* getting the transaction hash?

Reply:
we subscribe to transactions for the `fromAddress` right before making the `send` call. We are expecting a transaction after the `transactionHash` event is emitted if it makes sense